runtime/jailer.go

```diff
@@ -0,0 +1,299 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+
+	"github.com/firecracker-microvm/firecracker-go-sdk"
+)
+
+const (
+	jailerFolder        = "jail"
+	firecrackerBinName  = "firecracker"
+	kernelImageFileName = "kernel-image"
+	jailerHandlerName   = "firecracker-containerd-jail-handler"
+	runcConfigPath      = "/etc/containerd/firecracker-runc-config.json"
+
+	// JailingOn is used to signify whether or not jailing has been turned on
+	JailingOn = "on"
+	// JailingOff is used to signify that the jailer should not be used
+	JailingOff = "off"
+)
+
+// exposeFileToJail will inspect the given file, srcFilePath, and based on the
+// file type, proper handling will occur to ensure that the file is visible in
+// the jail.
+func exposeFileToJail(srcFilePath string, jailerRoot string) error {
```

Comment:
Is it possible to implement this by just doing a bind-mount (optionally a read-only bind mount) from the source to the jailer root? That should work for both normal files and block devices and simplify the implementation here a bit.